src/webgpu/api/operation/memory_sync/texture/same_subresource.spec.ts

```diff
@@ -42,15 +486,41 @@ g.test('rw')
         }
       })
   )
-  .unimplemented();
+  .fn(t => {
+    const helper = new TextureSyncTestHelper(t.device, {
+      usage:
+        GPUTextureUsage.COPY_DST |
+        kOpInfo[t.params.read.op].readUsage |
+        kOpInfo[t.params.write.op].writeUsage,
+    });
+    // [2] Use non-uniform texture value.
+    const texelValue1 = { R: 0, G: 1, B: 0, A: 1 } as const;
+    const texelValue2 = { R: 1, G: 0, B: 0, A: 1 } as const;
+
+    // Initialize the texture with something.
+    helper.performWriteOp({ op: 'write-texture', in: 'queue' }, texelValue1);
+    const readbackTexture = helper.performReadOp(t.params.read);
+    helper.ensureBoundary(t.params.boundary);
+    helper.performWriteOp(t.params.write, texelValue2);
+    helper.ensureSubmit();
+
+    // Contents should be the first value written, not the second.
+    t.expectSingleColor(readbackTexture, helper.kTextureFormat, {
+      size: [...helper.kTextureSize, 1],
+      exp: texelValue1,
+    });
+  });
 
 g.test('wr')
   .desc(
     `
     Perform a 'write' operation on a texture subresource, followed by a 'read' operation.
     Operations are separated by a 'boundary' (pass, encoder, queue-op, etc.).
     Test that the results are synchronized.
-    The read should see exactly the contents written by the previous write.`
+    The read should see exactly the contents written by the previous write.
+
+    - TODO: Finish implementation [1]
+    - TODO: Use non-uniform texture contents [2]`
```

Comment:
s/uniform/solid-color for clarity

Also since [2] is referenced in other tests as well, it should probably be in the file description.